tools/exporters/data_dictionary_builder.php

```diff
@@ -36,100 +36,102 @@
-    $DB,
-    $parameter_types,
-    array()
+    array(),
+    "Name"
 );
 
-//Delete in the parameter_type table relating to
-//parameter_type_category.Type = 'Instrument', without affecting other entries.
-//get parameter_type_category.Type ='Instrument' ParamenterTypeCategoryIDs
-getColumn(
-    "SELECT ParameterTypeCategoryID
-        FROM parameter_type_category
-        WHERE Type = :in",
-    $DB,
-    $instrumentParameterTypeCategoryIDs,
-    array('in' => 'Instrument')
+
+//Get all category IDs for instruments
+$instrumentParameterTypeCategoryIDs = $DB->pselectCol(
+    "SELECT ParameterTypeCategoryID 
+            FROM parameter_type_category 
+            WHERE Type = :in",
+    array("in" => "Instrument")
 );
 
+// DELETING all parameter_type data associated to instruments
 if (!empty($instrumentParameterTypeCategoryIDs)) {
-    // Prepare for the IN () SQL quandary
+    $instrumentParameterTypeIDString         = "";
+    $instrumentParameterTypeCategoryIDString = "";
+
+    //Prepare for the 'IN ()' SQL statement to get parameter_type values
+    //associated with the instrument categories
     $prepIn = prepareIn($instrumentParameterTypeCategoryIDs);
 
-    //get all 'Instrument' ParameterTypeIDs
-    getColumn(
+    //Get all types IDs related to instrument categories
+    $instrumentParameterTypeIDs = $DB->pselectCol(
         "SELECT ParameterTypeID
             FROM parameter_type_category_rel
             WHERE ParameterTypeCategoryID
-            IN (" . $prepIn['query'] . ")",
-        $DB,
-        $instrumentParameterTypeIDs,
-        $prepIn['array']
+            IN (" . $prepIn["query"] . ")",
+        $prepIn["array"]
     );
-    $instrumentParameterTypeIDString = implode(', ', $instrumentParameterTypeIDs);
-}
 
-if (!empty($instrumentParameterTypeIDString)) {
-    //delete all 'Instrument' entries from parameter_type_category_rel
-    $DB->run(
-        "DELETE FROM parameter_type_category_rel
-        WHERE ParameterTypeID in ($instrumentParameterTypeIDString)"
+    $instrumentParameterTypeIDString         = implode(
+        ", ",
+        $instrumentParameterTypeIDs
     );
-}
+    $instrumentParameterTypeCategoryIDString = implode(
+        ", ",
+        $instrumentParameterTypeCategoryIDs
+    );
+
+    if (!empty($instrumentParameterTypeIDString)) {
+        //delete all 'Instrument' entries from parameter_type_category_rel
+        //RUN function call required because of the IN clause
+        $DB->run(
+            "DELETE FROM parameter_type_category_rel
+        WHERE ParameterTypeID in ($instrumentParameterTypeIDString)"
+        );
+
+        //delete all 'Instrument' entries from parameter_type
+        //RUN function call required because of the IN clause
+        $DB->run(
+            "DELETE FROM parameter_type
+        WHERE ParameterTypeID IN ($instrumentParameterTypeIDString)"
+        );
+    }
 
-if (!empty($instrumentParameterTypeCategoryIDString)) {
```

Comment:
this variable `$instrumentParameterTypeCategoryIDString` was always empty since the string was never actually being generated... soooo bugfixed